src/OpenSpaceToolkit/Astrodynamics/Trajectory/State.cpp

```diff
@@ -47,38 +68,12 @@ State State::operator+(const State& aState) const
         throw ostk::core::error::runtime::Wrong("Instant");
     }
 
-    if (this->accessPosition().accessFrame() != aState.accessPosition().accessFrame())
+    if (this->frameSPtr_ != aState.frameSPtr_)
     {
-        throw ostk::core::error::runtime::Wrong("Position Frame");
+        throw ostk::core::error::runtime::Wrong("Frame");
     }
```

Comment:
Think it's worth it to also validate that the `coordinates_` vectors are the same length for the `+/-` operations?

Reply:
yup, done! 👍